### DISCUSSION TITLE
using manual tree crown delineation from ArcGIS rather than algo to obtain metrics

### DISCUSSION THREAD

**Discussion by lma210** [r-lidar/lidR#565] [Q&A]
Hi!
I normally use the Dalponte2016 algo to segment my trees and then use this to obtain some metrics (`stdmetrics_i`). 
I was wondering if it is possible to use individual tree boundaries that I draw manually on ArcGis rather than an algorithm (I draw the boundaries of 55 trees) to obtain metrics. My manual individual crown delineation gives me a  shapefile that is recognized as a `SpatialPolygonDataFrame`. But then, I don't know how to continue to obtain metrics. Did anyone try to do some metrics with a manual delineation from ArcGIS before? Thanks a lot!

This is le normal code I am using normally with the Dalponte algo:

```r
library(lidR)
library(raster)
library(rlas)
las <-  readLAS("SEFTON_clip1NGHCN.las", filter = "-drop_class 2") #reading lasfile already prepared with lastools (removed noise, ground, classification, height and normalisation
library(sf)
projection(las) <- 2193 # projection NZGD2000_NZTM2000, EPSG 2193 
chm_p2r_05 <- grid_canopy(las, res=0.5, algorithm=p2r(subcircle = 0.15)) #obtain CHM
kernel <- matrix(1,3,3)
chm_p2r_05_smoothed <- raster::focal(chm_p2r_05, w = kernel, fun = median, na.rm = TRUE) #filter
ttops_chm_p2r_05_smoothed <- find_trees(chm_p2r_05_smoothed, lmf(1)) #top of trees
algo <- dalponte2016(chm_p2r_05_smoothed, ttops_chm_p2r_05_smoothed)
lasL <- segment_trees(las, algo) 
crowns_metrics <- delineate_crowns(lasL, func = ~custom_tree_metrics(z = Z, i = Intensity))
head(crowns_metrics)
metricsR=data.frame(crowns_metrics)
write_xlsx(metricsR, 'metricsR.xlsx')
metrics1<- tree_metrics(lasL, .stdmetrics_i)
head(metrics1)
metrics1R=data.frame(metrics1)
write_xlsx(metrics1R, 'metrics1R.xlsx')
```

I would like to use the same code if possible, but with my manual crown data rather than Dalponte algo and the `segement_tree` function

```r
library(rgdal)
manualcrown <- readOGR(dsn = "D:/LiDAR/StatR/ManualCrown",layer = "ManualCrown")
plot(manualcrown)
manualcrown
```

**Comment by Jean-Romain** [comment]
Certainly doable with `merge_spatial` followed by `crown_metrics()`

**Reply by lma210** [reply]
Hi Jean-Romain,

It is what I was thinking yesterday. I tried this:

```r
MC<-merge_spatial(las,manualcrown)
```
but I got this error code:

```
x[i, ] is invalid
Attempting to make x[i, ] valid by zero-width buffering
Warning message:
In RGEOSUnaryPredFunc(spgeom, byid, "rgeos_isvalid") :
Ring Self-intersection at or near point 1571913.9341 5217544.3534000004
````

**Reply by Jean-Romain** [reply]
The error is self-explicit and has nothing to do with lidR. You shapefile contains invalid geometries

**Reply by lma210** [reply]
Hi Jean-Romain,

My bad. I will check this error on ArcGIS and see if it is working after fixing it. Thank you for your response and your time.

Leslie
